### PR TITLE
feat: add library-{esm,cjs,umd}

### PR DESCRIPTION
**Pull request by hardfist**
add example for building esm|cjs|umd library

### CODE REVIEW

**Review by SoonIter — APPROVED** [review]

File: rspack/library-cjs/package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "dev": "rspack serve"
```
Comment (fi3ework) [inline rspack/library-cjs/package.json:10]:
```suggestion
    "dev": "rspack build --watch"
```

File: rspack/library-esm/package.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "example-library-esm",
+  "version": "1.0.0",
+  "private": true,
+  "license": "MIT",
+  "sideEffects": ["**/*.css", "**/*.less", "**/*.scss"],
+  "main": "index.js",
+  "scripts": {
+    "build": "rspack build",
+    "dev": "rspack serve"
```
Comment (fi3ework) [inline rspack/library-esm/package.json:10]:
```suggestion
    "dev": "rspack build --watch"
```

File: rspack/library-umd/package.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "example-library-umd",
+  "version": "1.0.0",
+  "private": true,
+  "license": "MIT",
+  "sideEffects": ["**/*.css", "**/*.less", "**/*.scss"],
+  "main": "index.js",
+  "scripts": {
+    "build": "rspack build",
+    "dev": "rspack serve"
```
Comment (fi3ework) [inline rspack/library-umd/package.json:10]:
```suggestion
    "dev": "rspack build --watch"
```

File: rspack/library-cjs/package.json (hunk 1 above)
Comment (fi3ework) [reply rspack/library-cjs/package.json:10]: `serve` is useless in library mode?

**Review by fi3ework — APPROVED** [review]

**Pull request merged**